web3/middleware/filter.py

```diff
@@ -411,9 +373,9 @@ async def async_iter_latest_block(
     is_bounded_range = to_block is not None and to_block != "latest"
 
     while True:
-        latest_block = await w3.eth.block_number  # type: ignore
+        latest_block = await w3.eth.block_number
         # type ignored b/c is_bounded_range prevents unsupported comparison
```

Comment:
can remove `type ignored b/c` at type ignore removed

Reply:
This is separate. It's talking about the still-type-ignored `is_bounded_range` line below the comment.